### PR TITLE
Added curl request formatter

### LINKED ISSUE #160: logbook support log http request in CURL format
As a user, i want to print HttpClient request in CURL format which can easy to test by just execute at command line terminal, is this feature in your future plan？

### PR DESCRIPTION
**Pull request by whiskeysierra**
Fixes #160

### CODE REVIEW

**Comment by wyzssw** [comment]
It's very good ! Expecting new release .

File: logbook-core/src/main/java/org/zalando/logbook/CurlHttpLogFormatter.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            command.add(quote(body));
+        }
+
+        command.add(quote(request.getRequestUri()));
+
+        return command.stream().collect(joining(" "));
+    }
+
+    private static String quote(final String s) {
+        return "'" + escape(s) + "'";
```
Comment (AlexanderYastrebov) [inline logbook-core/src/main/java/org/zalando/logbook/CurlHttpLogFormatter.java]: In some cases quoting could be optional - e..g if you omit space after colon in headers and if there is no query string in url.

File: logbook-core/src/test/java/org/zalando/logbook/CurlHttpLogFormatterTest.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        final String correlationId = "c9408eaa-677d-11e5-9457-10ddb1ee7671";
+        final HttpRequest request = MockHttpRequest.create()
+                .withProtocolVersion("HTTP/1.0")
+                .withOrigin(Origin.REMOTE)
+                .withPath("/test")
+                .withQuery("limit=1")
+                .withHeaders(MockHeaders.of(
+                        "Accept", "application/json",
+                        "Content-Type", "text/plain"))
+                .withBodyAsString("Hello, world!");
```
Comment (AlexanderYastrebov) [inline logbook-core/src/test/java/org/zalando/logbook/CurlHttpLogFormatterTest.java]: Could you add test with json payload which itself contains strings with single and double quotes?

File: logbook-core/src/main/java/org/zalando/logbook/CurlHttpLogFormatter.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            values.forEach(value -> {
+                command.add("-H");
+                command.add(quote(header + ": " + value));
+            });
+        });
+
+        final String body = request.getBodyAsString();
+
+        if (!body.isEmpty()) {
+            command.add("-d");
```
Comment (AlexanderYastrebov) [inline logbook-core/src/main/java/org/zalando/logbook/CurlHttpLogFormatter.java]: I have checked 'Copy as cURL' in chrome dev tools:
1. They put URL in front of headers and body which makes sense
2. They use ` --data-binary` which also looks more proper.

File: logbook-core/src/main/java/org/zalando/logbook/CurlHttpLogFormatter.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    @Override
+    public String format(final Precorrelation<HttpRequest> precorrelation) throws IOException {
+        final HttpRequest request = precorrelation.getRequest();
+        final List<String> command = new ArrayList<>();
+
+        command.add("curl");
+        command.add("-v"); // TODO optional?
+
+        command.add("-X");
+        command.add(request.getMethod());
```
Comment (AlexanderYastrebov) [inline logbook-core/src/main/java/org/zalando/logbook/CurlHttpLogFormatter.java]: I believe most requests are GETs so I woudl add shortcut here to omit `-X GET`

Comment (AlexanderYastrebov) [inline logbook-core/src/main/java/org/zalando/logbook/CurlHttpLogFormatter.java]: Or better leave it for consistency

File: logbook-core/src/main/java/org/zalando/logbook/CurlHttpLogFormatter.java (hunk 1 above)
Comment (whiskeysierra) [inline logbook-core/src/main/java/org/zalando/logbook/CurlHttpLogFormatter.java]: I'd just always quote to be consistent. Also keeps the code simpler.

File: logbook-core/src/main/java/org/zalando/logbook/CurlHttpLogFormatter.java (hunk 3 above)
Comment (whiskeysierra) [inline logbook-core/src/main/java/org/zalando/logbook/CurlHttpLogFormatter.java]: Done.

File: logbook-core/src/test/java/org/zalando/logbook/CurlHttpLogFormatterTest.java (hunk 2 above)
Comment (whiskeysierra) [inline logbook-core/src/test/java/org/zalando/logbook/CurlHttpLogFormatterTest.java]: Done.

**Comment by AlexanderYastrebov** [comment]
:+1:

**Comment by whiskeysierra** [comment]
I'm still contemplating whether to put this into it's own module/artifact/repo outside of logbook, e.g. as a personal repo.

**Pull request merged**